src/extension/popup-page/index.tsx

```diff
@@ -99,13 +99,12 @@ function Popup() {
 function PopupWithProvider() {
     const preferDarkScheme = useMediaQuery('(prefers-color-scheme: dark)')
     const appearance = useValueRef(appearanceSettings)
+    const theme =
+        (preferDarkScheme && appearance === Appearance.default) || appearance === Appearance.dark
+            ? MaskbookDarkTheme
+            : MaskbookLightTheme
     return (
-        <ThemeProvider
-            theme={
-                (preferDarkScheme && appearance === Appearance.default) || appearance === Appearance.dark
-                    ? MaskbookDarkTheme
-                    : MaskbookLightTheme
-            }>
+        <ThemeProvider theme={Object.create(theme)}>
```

Comment:
Reason?

Reply:
A piece of a snippet from MUI codebase:

```ts
  const theme = React.useMemo(() => {
    const output = outerTheme === null ? localTheme : mergeOuterLocalTheme(outerTheme, localTheme);

    if (output != null) {
      output[nested] = outerTheme !== null;
    }

    return output;
  }, [localTheme, outerTheme]);
```

MUI always merge theme into the previous one. For some reason, this memo didn't update. Use `Object.create()` to force it update.

Reply:
It leads to a new theme object every-render

Reply:
It's intentional here.

Reply:
It's not mutating the previous theme https://github.com/mui-org/material-ui/blob/a3872cbb9cc95e91ab93875658d34fed1b878174/packages/material-ui-styles/src/ThemeProvider/ThemeProvider.js#L27 so I still not get why we need this hack here.

Reply:
A better workaround:
```
const theme = useMemo(
    () =>
        Object.create(
            (preferDarkScheme && appearance === Appearance.default) || appearance === Appearance.dark
                ? MaskbookDarkTheme
                : MaskbookLightTheme,
        ),
    [preferDarkScheme, appearance],
)
```

Reply:
Actually the 1st arg in `Object.create` means the _prototype_ of the new object. Why you need it on the prototype?

Reply:
It's OK with other methods to create a new object. Like `{...theme}`